src/bp/ui-shared/src/TreeView/index.tsx

```diff
@@ -0,0 +1,288 @@
+    node.isExpanded = isExpanded
+
+    props.onExpandToggle?.(node, isExpanded)
+  }
+
+  const handleNodeClick = (selectedNode: TreeNode<T>) => {
+    if (selectedNode.nodeData) {
+      props.onClick?.(selectedNode.nodeData)
+    }
+
+    traverseTree(nodes, node => {
+      if (node === selectedNode) {
+        if (props.highlightFolders || (!props.highlightFolders && node.type !== 'folder')) {
+          node.isSelected = !node.isSelected
+        }
+
+        if (!node.nodeData) {
+          node.isExpanded ? handleNodeExpansion(node, false) : handleNodeExpansion(node, true)
+        }
+      } else {
+        node.isSelected = false
+      }
+    })
+
+    forceUpdate()
+  }
+
+  const handleNodeDoubleClick = (selectedNode: TreeNode<T>) => {
+    if (selectedNode.nodeData) {
+      props.onDoubleClick?.(selectedNode.nodeData)
+    }
+  }
+
+  const handleContextMenu = (node: TreeNode<T>, _path, e) => {
+    if (!node.nodeData) {
+      return
+    }
+
+    e.preventDefault()
+
+    const contextMenu = props.onContextMenu?.(node.nodeData)
+    if (!contextMenu) {
+      return
+    }
+
+    ContextMenu.show(contextMenu, { left: e.clientX, top: e.clientY })
+  }
+
+  if (!nodes) {
+    return null
+  }
+
+  return (
+    <Tree
+      contents={nodes}
+      onNodeClick={node => handleNodeClick(node as TreeNode<T>)}
+      onNodeDoubleClick={node => handleNodeDoubleClick(node as TreeNode<T>)}
+      onNodeCollapse={node => handleNodeExpansion(node as TreeNode<T>, false)}
+      onNodeExpand={node => handleNodeExpansion(node as TreeNode<T>, true)}
+      onNodeContextMenu={(node, _p, e) => handleContextMenu(node as TreeNode<T>, _p, e)}
+      className={Classes.ELEVATION_0}
+    />
+  )
+}
+
+function splitPath<T>(
+  fullPath: string,
+  nodeData: T,
+  nodeRenderer: ElementRenderer<T>,
+  folderRenderer: ElementRenderer<string>
+) {
+  const splitPath = fullPath.split('/')
+  const nodeLabel = splitPath[splitPath.length - 1]
+  const parentFolders = splitPath.slice(0, splitPath.length - 1)
+
+  const folders: TreeNode<T>[] = []
+  const currentPath: string[] = []
+
+  for (const folder of parentFolders) {
+    currentPath.push(folder)
+    const { label, icon } = folderRenderer(folder)
+
+    folders.push({
+      id: folder,
+      label,
+      icon: icon || FOLDER_ICON,
+      fullPath: currentPath.join('/'),
+      type: 'folder',
+      childNodes: undefined
+    })
+  }
+
+  currentPath.push(nodeLabel)
+
+  const id = currentPath.join('/')
+  const { label, icon } = nodeRenderer(nodeData)
+
+  return {
+    folders,
+    leafNode: { id, label, icon: icon || DOCUMENT_ICON, fullPath: id, type: 'node' }
+  }
+}
+
+function buildTree<T>({
+  elements,
+  nodeRenderer,
+  filter,
+  folderRenderer,
+  postProcessing,
+  pathProps
+}: BuildTreeParams<T>): TreeNode<T>[] {
+  if (!elements) {
+    return []
+  }
+
+  if (!folderRenderer) {
```

Comment:
You could declare your function like this instead to avoid declaring folderRenderer
```
function buildTree<T>({
  elements,
  nodeRenderer,
  filter,
  folderRenderer = label => ({ label }),
  postProcessing,
  pathProps
}: BuildTreeParams<T>): TreeNode<T>[] {
```